modules/Movecraft/src/main/java/net/countercraft/movecraft/sign/NameSign.java

```diff
@@ -0,0 +1,54 @@
+package net.countercraft.movecraft.sign;
+
+import net.countercraft.movecraft.MovecraftLocation;
+import net.countercraft.movecraft.config.Settings;
+import net.countercraft.movecraft.craft.Craft;
+import net.countercraft.movecraft.events.CraftDetectEvent;
+import net.countercraft.movecraft.utils.ChatUtils;
+import org.bukkit.Material;
+import org.bukkit.World;
+import org.bukkit.block.Block;
+import org.bukkit.block.Sign;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.SignChangeEvent;
+import org.jetbrains.annotations.NotNull;
+
+public final class NameSign implements Listener {
+    String HEADER = "Name:";
+    @EventHandler
+    public void onCraftDetect(@NotNull CraftDetectEvent event){
+        Craft c = event.getCraft();
+        World w = c.getW();
+        for (MovecraftLocation location : event.getCraft().getHitBox()){
+            Block b = location.toBukkit(w).getBlock();
+            if (b.getType() == Material.SIGN_POST || b.getType() == Material.WALL_SIGN){
+                Sign s = (Sign) b.getState();
+                String name = "";
+                if (s.getLine(0).equalsIgnoreCase(HEADER)){
+                    if (s.getLine(1) != null){
```

Comment:
Sign line strings cannot be null. Instead, check if the line is the empty string